code-studio/src/js/components/Grid.jsx

```diff
@@ -68,7 +76,76 @@ var Grid = React.createClass({
     selectedRow: React.PropTypes.number,
     selectedCol: React.PropTypes.number,
     skin: React.PropTypes.string.isRequired,
-    onSelectionChange: React.PropTypes.func.isRequired
+    onSelectionChange: React.PropTypes.func.isRequired,
+    setCopiedCells: React.PropTypes.func.isRequired,
+  },
+
+  /**
+   * When drag begins, record that we are now dragging and where we
+   * started from.
+   */
+  beginDrag: function (row, col) {
+    this.setState({
+      dragging: true,
+      dragStart: {row: row, col: col},
+    });
+  },
+
+  /**
+   * As the mouse moves over the cells, if we are dragging then record
+   * the latest cell we've moved over so we can highlight all selected
+   * cells appropriately.
+   */
+  moveDrag: function (row, col) {
+    if (this.state && this.state.dragging) {
```

Comment:
You null-check state here, but not in `endDrag`.  Is this unneeded, or could it be avoided with a simple `getInitialState: () => {{}}` ?